test/modifiers/batch.test.js

```diff
@@ -0,0 +1,280 @@
+      var actual = [];
+
+      input
+        .pipe(batch({ count: 5 }))
+        .on('data', function (chunk) {
+          actual.push(chunk);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(actual).to.deep.equal(expectedData);
+          done();
+        });
+    });
+  });
+
+  describe('when both "time" and "count" are defined in options', function () {
+    it('writes chunks if count is met first before the defined time', function (done) {
+      var input = asyncStream.readable(_.times(10), { objectMode: true });
+      var chunks = 0;
+      var CHUNKS = 10;
+
+      input
+        .pipe(batch({
+          count: 1,
+          time: 5000
+        }))
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(1);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+    });
+
+    it('writes chunks if time is met first before the defined count', function (done) {
+      var CHUNKS = 10;
+      var EXPECTED = 5;
+      var reads = 0;
+      var input = asyncStream.readable(_.times(CHUNKS), { objectMode: true });
+
+      input
+        .pipe(batch({
+          count: 100,
+          time: 2
+        }))
+        .on('data', function (chunk) {
+          reads += 1;
+          expect(chunk).to.be.an('array').and.to.have.length.above(0);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(reads).to.equal(EXPECTED);
+          done();
+        });
+    });
+
+    it('successfully ends if there is no data written', function (done) {
+      var input = getReadableStream([], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch({
+          count: 2,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(false);
+          done();
+        });
+    });
+  });
+
+  function runSimpleTest(opts, done) {
+    var input = getReadableStream([1, 2, 3], {
+      objectMode: true
+    });
+
+    var expected = [[1], [2], [3]];
+    var actual = [];
+
+    input
+      .pipe(batch(opts))
+      .on('data', function (chunk) {
+        actual.push(chunk);
+      })
+      .on('error', done)
+      .on('end', function () {
+        expect(actual).to.deep.equal(expected);
+        done();
+      });
+  }
+
+  describe('when neither "time" nor "count" are defined in options', function () {
+    var expected = data.map(function (val) {
+      return [buff(val)];
+    });
+    var objExpected = objData.map(function (val) {
+      return [val];
+    });
+
+    basicTests({}, expected, objExpected);
+
+    it('writes chunks immediately when read', function (done) {
```

Comment:
So, it seems like this code expects each item to be wrapped in an array when no options are passed. That is the same behavior as setting `count = 1`. Shouldn't there be different behavior for count of 0 and 1?